mdsobjects/cpp/mdstreeobjects.cpp

```diff
@@ -1661,7 +1681,337 @@ StringArray *TreeNodeArray::getUsage()
 
 
 ////////////////End TreeNodeArray methods/////////////////////
+////////////////TreeNodeThinClient methods ///////////////////
+
+static const char *convertNciItm(int nciItm)
+{
+    switch(nciItm) {
+      case NciTIME_INSERTED: 		return "\'TIME_INSERTED\'";
+      case NciOWNER_ID: 		return "\'OWNER_ID\'"; 
+      case NciCLASS:			return "\'CLASS\'";
+      case NciDTYPE:			return "\'DTYPE\'";
+      case NciLENGTH:			return "\'LENGTH\'";
+      case NciSTATUS:			return "\'STATUS\'";
+      case NciCONGLOMERATE_ELT:		return "\'CONGLOMERATE_ELT\'";
+      case NciGET_FLAGS:		return "\'GET_FLAGS\'";
+      case NciNODE_NAME:		return "\'NODE_NAME\'";
+      case NciPATH:			return "\'PATH\'";
+      case NciDEPTH:			return "\'DEPTH\'";
+      case NciPARENT:			return "\'PARENT\'";
+      case NciBROTHER:			return "\'BROTHER\'";
+      case NciMEMBER:			return "\'MEMBER\'";
+      case NciCHILD:			return "\'CHILD\'";
+      case NciPARENT_RELATIONSHIP:	return "\'PARENT_RELATIONSHIP\'";
+      case NciK_IS_CHILD:		return "\'IS_CHILD\'";
+      case NciK_IS_MEMBER:		return "\'IS_MEMBER\'";
+      case NciCONGLOMERATE_NIDS:	return "\'CONGLOMERATE_NIDS\'";
+      case NciORIGINAL_PART_NAME:	return "\'ORIGINAL_PART_NAME\'";
+      case NciNUMBER_OF_MEMBERS:	return "\'NUMBER_OF_MEMBERS\'";
+      case NciNUMBER_OF_CHILDREN :	return "\'NUMBER_OF_CHILDREN\'";
+      case NciMEMBER_NIDS:		return "\'MEMBER_NIDS\'";
+      case NciCHILDREN_NIDS:		return "\'CHILDREN_NIDS\'";
+      case NciFULLPATH:			return "\'FULLPATH\'";
+      case NciMINPATH:			return "\'MINPATH\'";
+      case NciUSAGE:			return "\'USAGE\'";
+      default:				throw MdsException("Unexpected NCI item in TreeNodeThinClient");
+    }
+}
+
+TreeNodeThinClient::TreeNodeThinClient(int nid, Connection *connection, Data *units, Data *error, Data *help, Data *validation)
+{
+    this->tree = 0;
+    this->nid = nid;
+    this->connection = connection;
+    clazz = CLASS_S;
+    dtype = DTYPE_NID;
+    setAccessory(units, error, help, validation);
+}
+
+std::string  TreeNodeThinClient::getNciString(int itm)
+{
+    char expr[64];
+    sprintf(expr, "GETNCI(%d,%s)", nid, convertNciItm(itm));
+    Data *retStringData = connection->get(expr);
+    if(!retStringData)
+	throw MdsException("Error in Remote evaluation of getnci");
+    
+    AutoString as(retStringData->getString());
+//    char *retStringPtr = retStringData->getString();
+//    std::string retString(retStringPtr);
+//    delete [] retStringPtr;
+    MDSplus::deleteData(retStringData);
+    return as.string;
+}
+
+char TreeNodeThinClient::getNciChar(int itm)
```

Comment:
Complete example:

char TreeNodeThinClient::getNciChar(int itm)
{
    char expr[64];
    sprintf(expr, "GETNCI(%d,%s)", nid, convertNciItm(itm));
    AutoData<Data> d(connection->get(expr));
    if (!d.get())
        throw MdsException("Error in Remote evaluation of getnci");
    return d->getByte();
}